try to add fmp senate trading data api

markqiu opened the pull request:
# Pull Request OpenBB

Please go to the `Preview` tab and select the appropriate PR sub-template:

* [OpenBB Platform](?expand=1&template=platform_pull_request_template.md)
* [OpenBB Platform CLI](?expand=1&template=terminal_pull_request_template.md)
* [OpenBB Developers](?expand=1&template=obb_developer_pull_request_template.md)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OpenBB-finance/OpenBB?pullRequest=6761) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/OpenBB-finance/OpenBB?pullRequest=6761) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: deeleeramone<br/>:white_check_mark: montezdesousa<br/>:white_check_mark: piiq<br/>:white_check_mark: hjoaquim<br/>:x: markqiu<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OpenBB-finance/OpenBB?pullRequest=6761) it.</sub>

deeleeramone commented:
Hi, thanks for the PR!

This is off to a good start, and looks good so far.  Let's discuss some of the higher-level implementation details.

These are just my opinions on how I would like to consume this type of data.

- "House" and "Senate" should be a single function, where excluding a symbol returns the "RSS" version of each.
  - Something like, `chamber = Literal["house", "senate", "all"]`, where "all" would make 2 calls and combine results.

This would consolidate four models down to one - some name like, `GovernmentTrades`.

- There should be optional query params for start/end dates and the representative_name.
  - If the provider doesn't offer these params, we can filter the data dump post-request.

The returned fields should consolidate as well, mapping to something that makes sense for everything and drops redundancies.

The only fields in the standard data model that should be mandatory are:
- "symbol"
- "date" (just the mapped "disclosureDate/dateReceived")
- "transaction_date"
- "representative"

Everything else would be `Optional` with a default of `None`.

Standard models need to be usable for sources, not just FMP or any single provider, and we can't assume all fields are always returned from every provider. 

### House

```json
{
"disclosureYear": "2024",
"disclosureDate": "2024-10-03",
"transactionDate": "2024-09-12",
"owner": "Joint",
"ticker": "AAPL",
"assetDescription": "Apple Inc. - Common Stock",
"type": "Purchase",
"amount": "$1,001 - $15,000",
"representative": "Josh Gottheimer",
"district": "NJ05",
"link": "https://disclosures-clerk.house.gov/public_disc/ptr-pdfs/2024/20025953.pdf",
"capitalGainsOver200USD": "False"
}
```

### Senate
```json
{
"firstName": "Shelley",
"lastName": "Moore Capito",
"office": "Shelley Moore Capito",
"link": "https://efdsearch.senate.gov/search/view/ptr/1489e003-c0e6-4370-9068-7c42946e5555/",
"dateRecieved": "2024-10-04",
"transactionDate": "2024-09-20",
"owner": "Spouse",
"assetDescription": "Apple Inc",
"assetType": "Stock",
"type": "Sale (Partial)",
"amount": "$1,001 - $15,000",
"comment": "--",
"symbol": "AAPL"
}
```

"ticker" in House and "symbol" in Senate. This should be "symbol", and you did map that accordingly already. Let's take that same approach to narrow the difference between other fields.

- "disclosureYear" is redundant, we can drop that.
- "dateReceived" and "disclosureDate" are the same thing, and that could be named "date".
- From the Senate, we don't need "firstName", "lastName", and "office" - assuming 'office' is always the full name -  would be the same as "representative" in the House data.


I might be missing a few details, but that's the general idea.

As-is now, you have created the connections in the Provider Interface, but they do not yet exist within the Router. See the documentation section [here](https://docs.openbb.co/platform/user_guides/add_endpoint_to_existing_provider#create-router-endpoint) for a sample.

Thoughts?

mmistroni commented:
@markqiu  you commented on one of my old PR (which was dropped) and looks like you are quicker than me here :) as i have a similar PR
Shall we help each other out? happy to fork your repo and use your branch.
Here's my notes on the FMP , bearing in mind what @deeleeramone mentioned and i might repeat what you already know, so bear with me

FMP has 4 APIS (they had 3 up to a week ago)
- senate-trading-rss. this covers 'senate trades'. it's driven by a 'page' parameter which is 'unbounded'. page0 - as per fmp support - returns the last 100 days worth of trades. IMHO, very useful as a 'daily' check, and perhaps its' worth checking page0 and page1 for the last 200 days worth of trades. 
- senate-disclosure-rss (this i did not know), presumably does the same.but is badly named, as it points to HOR.
- senate-trading  - queryable by ticker. historical
- senate-disclosures - queryable by ticker.  historical

Personally, to keep it simple, i'd expose 2 endpoiints:
- congress_trades_historical, queryable by ticker/dates which points to the latter 2 APIs,
- congress_trades_latest which points to the first two , . No query inputs, but retrieval will use perhaps with page0 and page1 to get the last 200 days worth of trades

That'll keep it simple so that we dont have to flip between 4 different APIs depending on the query parameters
if i were to search eg if NVDA was trades by congress in last 3 months, i'll use the last 2 FMP api mentioned
If i were to search on 'what congress has traded yesterday', which is driven by no parameters, i'll go for the first 2 apis

Also, as per router, in my branch i was planning to put it under equity.discovery, alongside filings?

let me know what you think

mmistroni commented:
also, forgot one more comment. i tried to merge the two 'historical' api calls into one by using 'justs' the common fields.. here's my cb

# include both hor and senate
await asyncio.gather(*[get_one(ep) for ep in ['senate-trading', 'senate-disclosure']])

and here's my transform data

@staticmethod
    def transform_data(
        query: FMPCongressTradingHistoricalQueryParams, data: List[Dict], **kwargs: Any
    ) -> List[FMPCongressTradingHistoricalData]:
        """Return the transformed data."""
        def _map_hor(item):
            return dict(ticker=item.get('ticker', ''),
                        filer=item.get('representative' ,''),
                        transactionDate=item.get('transactionDate', ''),
                        type=item.get('type', ''),
                        amount=item.get('amount', ''),
                        filingDate=item.get('disclosureDate'),
                        source='HouseOfRepresentative'
                        )

        def _map_senate(item):
            return dict(ticker=item.get('ticker', ''),
                        filer=f"{item.get('firstName' ,'')},{item.get('lastName' ,'')}",
                        transactionDate=item.get('transactionDate', ''),
                        type=item.get('type', ''),
                        amount=item.get('amount', ''),
                        filingDate=item.get('dateRecieved'),
                        source='Senate'
                        )

        results: List[FMPCongressTradingHistoricalData] = []
        for item in data:
            new_item = _map_hor(item) if 'disclosureYear' in item.keys() else _map_senate(item)
            results.append(FMPCongressTradingHistoricalData.model_validate(new_item))
        return results

guess we'll have to do same for rss's.... (didnt see the hor rss til today.... but  i think senate guys make more money than HORs :)

mmistroni commented:
and, sorry to spam , we  might be able to source houseofrep ourselves... @deeleeramone  passed me a notebook some time ago for sourcing senate data.. i reused it for HOR (as data is stored in PDFs based on year) so it's not too hard
I'll send you the senate too, that is harder as it uses a hidden session id and tracks your IP as well

markqiu commented:
> @markqiu you commented on one of my old PR (which was dropped) and looks like you are quicker than me here :) as i have a similar PR Shall we help each other out? happy to fork your repo and use your branch. Here's my notes on the FMP , bearing in mind what @deeleeramone mentioned and i might repeat what you already know, so bear with me
> 
> FMP has 4 APIS (they had 3 up to a week ago)
> 
> * senate-trading-rss. this covers 'senate trades'. it's driven by a 'page' parameter which is 'unbounded'. page0 - as per fmp support - returns the last 100 days worth of trades. IMHO, very useful as a 'daily' check, and perhaps its' worth checking page0 and page1 for the last 200 days worth of trades.
> * senate-disclosure-rss (this i did not know), presumably does the same.but is badly named, as it points to HOR.
> * senate-trading  - queryable by ticker. historical
> * senate-disclosures - queryable by ticker.  historical
> 
> Personally, to keep it simple, i'd expose 2 endpoiints:
> 
> * congress_trades_historical, queryable by ticker/dates which points to the latter 2 APIs,
> * congress_trades_latest which points to the first two , . No query inputs, but retrieval will use perhaps with page0 and page1 to get the last 200 days worth of trades
> 
> That'll keep it simple so that we dont have to flip between 4 different APIs depending on the query parameters if i were to search eg if NVDA was trades by congress in last 3 months, i'll use the last 2 FMP api mentioned If i were to search on 'what congress has traded yesterday', which is driven by no parameters, i'll go for the first 2 apis
> 
> Also, as per router, in my branch i was planning to put it under equity.discovery, alongside filings?
> 
> let me know what you think

That's great, I'll take the time to go through all the replies carefully (I've been quite busy these days).

mmistroni commented:
yes,no w. and pls let me  know if i can fork out your repo as well so i can
help out?
thanks and kind regards
 Marco

On Sun, Oct 27, 2024 at 12:39 PM 邱承 ***@***.***> wrote:

> @markqiu <https://github.com/markqiu> you commented on one of my old PR
> (which was dropped) and looks like you are quicker than me here :) as i
> have a similar PR Shall we help each other out? happy to fork your repo and
> use your branch. Here's my notes on the FMP , bearing in mind what
> @deeleeramone <https://github.com/deeleeramone> mentioned and i might
> repeat what you already know, so bear with me
>
> FMP has 4 APIS (they had 3 up to a week ago)
>
>    - senate-trading-rss. this covers 'senate trades'. it's driven by a
>    'page' parameter which is 'unbounded'. page0 - as per fmp support - returns
>    the last 100 days worth of trades. IMHO, very useful as a 'daily' check,
>    and perhaps its' worth checking page0 and page1 for the last 200 days worth
>    of trades.
>    - senate-disclosure-rss (this i did not know), presumably does the
>    same.but is badly named, as it points to HOR.
>    - senate-trading - queryable by ticker. historical
>    - senate-disclosures - queryable by ticker. historical
>
> Personally, to keep it simple, i'd expose 2 endpoiints:
>
>    - congress_trades_historical, queryable by ticker/dates which points
>    to the latter 2 APIs,
>    - congress_trades_latest which points to the first two , . No query
>    inputs, but retrieval will use perhaps with page0 and page1 to get the last
>    200 days worth of trades
>
> That'll keep it simple so that we dont have to flip between 4 different
> APIs depending on the query parameters if i were to search eg if NVDA was
> trades by congress in last 3 months, i'll use the last 2 FMP api mentioned
> If i were to search on 'what congress has traded yesterday', which is
> driven by no parameters, i'll go for the first 2 apis
>
> Also, as per router, in my branch i was planning to put it under
> equity.discovery, alongside filings?
>
> let me know what you think
>
> That's great, I'll take the time to go through all the replies carefully
> (I've been quite busy these days).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2439999666>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPNCDULS72SUORDX7QQXPDZ5TNGVAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMZZHE4TSNRWGY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

markqiu commented:
Regarding the data about the Senate and the House of Representatives, there are differing recommendations from the @deeleeramone  and @mmistroni.

deeleeramone's recommendation: Write a fetcher with parameters
```python
params:
chamber = Literal["house", "senate", "all"]
symbol
```
to control the return of different results.
- If `chamber` is set to `house`:
  - Passing `symbol` returns historical data for the House of Representatives.
  - Not passing `symbol` returns the latest few hundred records for the House of Representatives.
- If `chamber` is set to `senate`:
  - The same as above but for the Senate data.
- If `chamber` is set to `all`:
  - Passing `symbol` returns integrated historical data for both the House of Representatives and the Senate.
  - Not passing `symbol` returns the latest few hundred integrated records for both the House of Representatives and the Senate.

mmistroni’s recommendation: Write two fetchers `congress_trades_historical` and `congress_trades_latest`.
```python
params:
chamber = Literal["house", "senate", "all"]
symbol
```
- `congress_trades_historical`: Returns historical trades.
- `congress_trades_latest`: Returns the latest trades.
- Passing `chamber` and `symbol` filters the latest results for a specific symbol (this effect is not achieved by the official method).

I lean towards using the first approach because I haven't noticed any difference between `historical` and `latest` in the second approach. What do you think about this issue?
@deeleeramone @mmistroni

markqiu commented:
> yes,no w. and pls let me know if i can fork out your repo as well so i can help out? thanks and kind regards Marco
> […](#)
> On Sun, Oct 27, 2024 at 12:39 PM 邱承 ***@***.***> wrote: @markqiu <https://github.com/markqiu> you commented on one of my old PR (which was dropped) and looks like you are quicker than me here :) as i have a similar PR Shall we help each other out? happy to fork your repo and use your branch. Here's my notes on the FMP , bearing in mind what @deeleeramone <https://github.com/deeleeramone> mentioned and i might repeat what you already know, so bear with me FMP has 4 APIS (they had 3 up to a week ago) - senate-trading-rss. this covers 'senate trades'. it's driven by a 'page' parameter which is 'unbounded'. page0 - as per fmp support - returns the last 100 days worth of trades. IMHO, very useful as a 'daily' check, and perhaps its' worth checking page0 and page1 for the last 200 days worth of trades. - senate-disclosure-rss (this i did not know), presumably does the same.but is badly named, as it points to HOR. - senate-trading - queryable by ticker. historical - senate-disclosures - queryable by ticker. historical Personally, to keep it simple, i'd expose 2 endpoiints: - congress_trades_historical, queryable by ticker/dates which points to the latter 2 APIs, - congress_trades_latest which points to the first two , . No query inputs, but retrieval will use perhaps with page0 and page1 to get the last 200 days worth of trades That'll keep it simple so that we dont have to flip between 4 different APIs depending on the query parameters if i were to search eg if NVDA was trades by congress in last 3 months, i'll use the last 2 FMP api mentioned If i were to search on 'what congress has traded yesterday', which is driven by no parameters, i'll go for the first 2 apis Also, as per router, in my branch i was planning to put it under equity.discovery, alongside filings? let me know what you think That's great, I'll take the time to go through all the replies carefully (I've been quite busy these days). — Reply to this email directly, view it on GitHub <[#6761 (comment)](https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2439999666)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACPNCDULS72SUORDX7QQXPDZ5TNGVAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMZZHE4TSNRWGY> . You are receiving this because you commented.Message ID: ***@***.***>

I will try to make a branch for you. I have a big one now, need to do some cut.

mmistroni commented:
@deeleramone suggestion will always supersede mines :)  , happy to go with
them,   but having used the 'latest disclosures'  from FMP for last year or
so, i could only see two possible usecases

1 - congress_historical_trades - search for historical trades for a ticker
(this will be irrespective of senate or hor.).

     params:

                      ['ticker', start_date:Optional, end_date:Optional].

this would cover usecase where you want to know whom in the congress trades
a particular ticker historically.Useful for historical analysis if you want
to see the impact of congress buying a particular stock. which could also
be cross-checked with Government contracts to see who'll benefit from this.

2 - congress_latest:  Will return last 100 days worth of congress trades
(for both senate and hor)

            params:[]

this would cover usecase where you want to know what congress trades
recently.

IMHO would make it for an easier implementation, discarding all the
possible switches you'll have to do depending on input parameters

But, once again, perhaps i am focused on my specific usecases and don't see
the big picture....

my 2 cents.. happy to go with yours and @deeleramone

pls let me know if i can help at all

kind regards

Marco





On Wed, 30 Oct 2024, 11:19 邱承, ***@***.***> wrote:

> yes,no w. and pls let me know if i can fork out your repo as well so i can
> help out? thanks and kind regards Marco
> … <#m_7874944281260813783_m_1207098040333221458_>
> On Sun, Oct 27, 2024 at 12:39 PM 邱承 *@*.*> wrote: @markqiu
> <https://github.com/markqiu> https://github.com/markqiu
> <https://github.com/markqiu> you commented on one of my old PR (which was
> dropped) and looks like you are quicker than me here :) as i have a similar
> PR Shall we help each other out? happy to fork your repo and use your
> branch. Here's my notes on the FMP , bearing in mind what @deeleeramone
> <https://github.com/deeleeramone> https://github.com/deeleeramone
> <https://github.com/deeleeramone> mentioned and i might repeat what you
> already know, so bear with me FMP has 4 APIS (they had 3 up to a week ago)
> - senate-trading-rss. this covers 'senate trades'. it's driven by a 'page'
> parameter which is 'unbounded'. page0 - as per fmp support - returns the
> last 100 days worth of trades. IMHO, very useful as a 'daily' check, and
> perhaps its' worth checking page0 and page1 for the last 200 days worth of
> trades. - senate-disclosure-rss (this i did not know), presumably does the
> same.but is badly named, as it points to HOR. - senate-trading - queryable
> by ticker. historical - senate-disclosures - queryable by ticker.
> historical Personally, to keep it simple, i'd expose 2 endpoiints: -
> congress_trades_historical, queryable by ticker/dates which points to the
> latter 2 APIs, - congress_trades_latest which points to the first two , .
> No query inputs, but retrieval will use perhaps with page0 and page1 to get
> the last 200 days worth of trades That'll keep it simple so that we dont
> have to flip between 4 different APIs depending on the query parameters if
> i were to search eg if NVDA was trades by congress in last 3 months, i'll
> use the last 2 FMP api mentioned If i were to search on 'what congress has
> traded yesterday', which is driven by no parameters, i'll go for the first
> 2 apis Also, as per router, in my branch i was planning to put it under
> equity.discovery, alongside filings? let me know what you think That's
> great, I'll take the time to go through all the replies carefully (I've
> been quite busy these days). — Reply to this email directly, view it on
> GitHub <#6761 (comment)
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2439999666>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ACPNCDULS72SUORDX7QQXPDZ5TNGVAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMZZHE4TSNRWGY
> <https://github.com/notifications/unsubscribe-auth/ACPNCDULS72SUORDX7QQXPDZ5TNGVAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMZZHE4TSNRWGY>
> . You are receiving this because you commented.Message ID: @.*>
>
> I will try to make a branch for you. I have a big one now, need to do some
> cut.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2446684841>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPNCDT3KXMMS2SVOT62W2LZ6C6F3AVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINBWGY4DIOBUGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

markqiu commented:
Hello,

Based on our previous discussions, my colleague and I attempted to integrate the Senate API and also tried to incorporate FMP's valuation data into OpenBB. Could you please inform us of any issues that have arisen? Thank you.

Best regards,
Markqiu

@deeleeramone @mmistroni

mmistroni commented:
am not OBB .
very good job!!! i would have kept though 4 branches, one per feature as
OBBs are quite busy in review (government trading, ratings, dcf, 13f).
guess @deeleramonen wil come back to you at some poinit

rgds

On Tue, Nov 19, 2024 at 3:00 AM 邱承 ***@***.***> wrote:

> Hello,
>
> Based on our previous discussions, my colleague and I attempted to
> integrate the Senate API and also tried to incorporate FMP's valuation data
> into OpenBB. Could you please inform us of any issues that have arisen?
> Thank you.
>
> Best regards,
> Markqiu
>
> @deeleeramone <https://github.com/deeleeramone> @mmistroni
> <https://github.com/mmistroni>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2484606948>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPNCDV7SQE6TUK2VGZXXRL2BKSWJAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGYYDMOJUHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mmistroni commented:
also my personal 2 cents on the 13f...
personally dont think  FMP EP very useful as you'd need to know CIK
They also dont  mention what is the filing period.
Institutional investors file these form 'at whim' and they can also do a
filing for a previous year as they are not obliged to do a 13f at the end
of each quarter - i believe

what could be useful IMHO would be the daily 13hf filings or the quarterly
13hf filings, perhaps filtered only by filing related to last quarter or
perhaps last 3 quarters

This data is publicly available via edgar -  im scraping it daily -  and
it's stored in txt / XML.
so perhaps we can bypass FMP and offer a better endpoint?

what do you think?

PS  i think there's an edgar api that @deeleramone suggested me some time
ago, it's called edgartools and will help a lot - rather than doing what i
am doing, scraping txt file from edgar archives



On Tue, Nov 19, 2024 at 9:46 AM Sofia’s World ***@***.***> wrote:

> am not OBB .
> very good job!!! i would have kept though 4 branches, one per feature as
> OBBs are quite busy in review (government trading, ratings, dcf, 13f).
> guess @deeleramonen wil come back to you at some poinit
>
> rgds
>
> On Tue, Nov 19, 2024 at 3:00 AM 邱承 ***@***.***> wrote:
>
>> Hello,
>>
>> Based on our previous discussions, my colleague and I attempted to
>> integrate the Senate API and also tried to incorporate FMP's valuation data
>> into OpenBB. Could you please inform us of any issues that have arisen?
>> Thank you.
>>
>> Best regards,
>> Markqiu
>>
>> @deeleeramone <https://github.com/deeleeramone> @mmistroni
>> <https://github.com/mmistroni>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2484606948>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACPNCDV7SQE6TUK2VGZXXRL2BKSWJAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGYYDMOJUHA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

markqiu commented:
> also my personal 2 cents on the 13f... personally dont think FMP EP very useful as you'd need to know CIK They also dont mention what is the filing period. Institutional investors file these form 'at whim' and they can also do a filing for a previous year as they are not obliged to do a 13f at the end of each quarter - i believe what could be useful IMHO would be the daily 13hf filings or the quarterly 13hf filings, perhaps filtered only by filing related to last quarter or perhaps last 3 quarters This data is publicly available via edgar - im scraping it daily - and it's stored in txt / XML. so perhaps we can bypass FMP and offer a better endpoint? what do you think? PS i think there's an edgar api that @deeleramone suggested me some time ago, it's called edgartools and will help a lot - rather than doing what i am doing, scraping txt file from edgar archives
> […](#)
> On Tue, Nov 19, 2024 at 9:46 AM Sofia’s World ***@***.***> wrote: am not OBB . very good job!!! i would have kept though 4 branches, one per feature as OBBs are quite busy in review (government trading, ratings, dcf, 13f). guess @deeleramonen wil come back to you at some poinit rgds On Tue, Nov 19, 2024 at 3:00 AM 邱承 ***@***.***> wrote: > Hello, > > Based on our previous discussions, my colleague and I attempted to > integrate the Senate API and also tried to incorporate FMP's valuation data > into OpenBB. Could you please inform us of any issues that have arisen? > Thank you. > > Best regards, > Markqiu > > @deeleeramone <https://github.com/deeleeramone> @mmistroni > <https://github.com/mmistroni> > > — > Reply to this email directly, view it on GitHub > <[#6761 (comment)](https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2484606948)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/ACPNCDV7SQE6TUK2VGZXXRL2BKSWJAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGYYDMOJUHA> > . > You are receiving this because you were mentioned.Message ID: > ***@***.***> >

Thank you for your advices. We will discuss them and split our PR to 4 branches, one per feature.

mmistroni commented:
Hey.. don't want to disappoint you guys but I thk bob has already a form13f
here

https://docs.openbb.co/platform/reference/equity/ownership


Hth

On Wed, 20 Nov 2024, 01:54 邱承, ***@***.***> wrote:

> also my personal 2 cents on the 13f... personally dont think FMP EP very
> useful as you'd need to know CIK They also dont mention what is the filing
> period. Institutional investors file these form 'at whim' and they can also
> do a filing for a previous year as they are not obliged to do a 13f at the
> end of each quarter - i believe what could be useful IMHO would be the
> daily 13hf filings or the quarterly 13hf filings, perhaps filtered only by
> filing related to last quarter or perhaps last 3 quarters This data is
> publicly available via edgar - im scraping it daily - and it's stored in
> txt / XML. so perhaps we can bypass FMP and offer a better endpoint? what
> do you think? PS i think there's an edgar api that @deeleramone suggested
> me some time ago, it's called edgartools and will help a lot - rather than
> doing what i am doing, scraping txt file from edgar archives
> … <#m_3289162718268362516_m_664976961443849155_>
> On Tue, Nov 19, 2024 at 9:46 AM Sofia’s World *@*.*> wrote: am not OBB .
> very good job!!! i would have kept though 4 branches, one per feature as
> OBBs are quite busy in review (government trading, ratings, dcf, 13f).
> guess @deeleramonen wil come back to you at some poinit rgds On Tue, Nov
> 19, 2024 at 3:00 AM 邱承 @.*> wrote: > Hello, > > Based on our previous
> discussions, my colleague and I attempted to > integrate the Senate API and
> also tried to incorporate FMP's valuation data > into OpenBB. Could you
> please inform us of any issues that have arisen? > Thank you. > > Best
> regards, > Markqiu > > @deeleeramone <https://github.com/deeleeramone>
> https://github.com/deeleeramone @mmistroni <https://github.com/mmistroni>
> > https://github.com/mmistroni > > — > Reply to this email directly, view
> it on GitHub > <#6761 (comment)
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2484606948>>,
> > or unsubscribe >
> https://github.com/notifications/unsubscribe-auth/ACPNCDV7SQE6TUK2VGZXXRL2BKSWJAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGYYDMOJUHA
> > . > You are receiving this because you were mentioned.Message ID: > *@*.***>
> >
>
> Thank you for your advices. We will discuss them and split our PR to 4
> branches, one per feature.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2487152667>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPNCDQCS2OYHCZJLVR3MWT2BPTT5AVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBXGE2TENRWG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

deeleeramone commented:
> Hello,
> 
> Based on our previous discussions, my colleague and I attempted to integrate the Senate API and also tried to incorporate FMP's valuation data into OpenBB. Could you please inform us of any issues that have arisen? If any modifications are needed, we are more than willing to assist. Thank you.
> 
> Best regards, Markqiu
> 
> @deeleeramone @mmistroni

Hi, @markqiu, can you please scope the PR to the original intention?

I would still like to have all the representative trading endpoints consolidated as a single function. We don't want to create standard models and endpoints that can only be used for fetching data from FMP.

The DCF endpoints are not particularly of interest - they are generic black box calculations that have no configurability or transparency.  A DCF needs to be adaptable and modular, specific for a company.

markqiu commented:
> Hey.. don't want to disappoint you guys but I thk bob has already a form13f here https://docs.openbb.co/platform/reference/equity/ownership Hth
> […](#)
> On Wed, 20 Nov 2024, 01:54 邱承, ***@***.***> wrote: also my personal 2 cents on the 13f... personally dont think FMP EP very useful as you'd need to know CIK They also dont mention what is the filing period. Institutional investors file these form 'at whim' and they can also do a filing for a previous year as they are not obliged to do a 13f at the end of each quarter - i believe what could be useful IMHO would be the daily 13hf filings or the quarterly 13hf filings, perhaps filtered only by filing related to last quarter or perhaps last 3 quarters This data is publicly available via edgar - im scraping it daily - and it's stored in txt / XML. so perhaps we can bypass FMP and offer a better endpoint? what do you think? PS i think there's an edgar api that @deeleramone suggested me some time ago, it's called edgartools and will help a lot - rather than doing what i am doing, scraping txt file from edgar archives … <#m_3289162718268362516_m_664976961443849155_> On Tue, Nov 19, 2024 at 9:46 AM Sofia’s World *@*.*> wrote: am not OBB . very good job!!! i would have kept though 4 branches, one per feature as OBBs are quite busy in review (government trading, ratings, dcf, 13f). guess @deeleramonen wil come back to you at some poinit rgds On Tue, Nov 19, 2024 at 3:00 AM 邱承 @.*> wrote: > Hello, > > Based on our previous discussions, my colleague and I attempted to > integrate the Senate API and also tried to incorporate FMP's valuation data > into OpenBB. Could you please inform us of any issues that have arisen? > Thank you. > > Best regards, > Markqiu > > @deeleeramone <https://github.com/deeleeramone> https://github.com/deeleeramone @mmistroni <https://github.com/mmistroni> > https://github.com/mmistroni > > — > Reply to this email directly, view it on GitHub > <#6761 (comment) <[#6761 (comment)](https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2484606948)>>, > or unsubscribe > https://github.com/notifications/unsubscribe-auth/ACPNCDV7SQE6TUK2VGZXXRL2BKSWJAVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGYYDMOJUHA > . > You are receiving this because you were mentioned.Message ID: > *@*.***> > Thank you for your advices. We will discuss them and split our PR to 4 branches, one per feature. — Reply to this email directly, view it on GitHub <[#6761 (comment)](https://github.com/OpenBB-finance/OpenBB/pull/6761#issuecomment-2487152667)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACPNCDQCS2OYHCZJLVR3MWT2BPTT5AVCNFSM6AAAAABP2ANHFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBXGE2TENRWG4> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Yes, we just added a fmp provider for it.

markqiu commented:
> > Hello,
> > Based on our previous discussions, my colleague and I attempted to integrate the Senate API and also tried to incorporate FMP's valuation data into OpenBB. Could you please inform us of any issues that have arisen? If any modifications are needed, we are more than willing to assist. Thank you.
> > Best regards, Markqiu
> > @deeleeramone @mmistroni
> 
> Hi, @markqiu, can you please scope the PR to the original intention?
> 
> I would still like to have all the representative trading endpoints consolidated as a single function. We don't want to create standard models and endpoints that can only be used for fetching data from FMP.
> 
> The DCF endpoints are not particularly of interest - they are generic black box calculations that have no configurability or transparency. A DCF needs to be adaptable and modular, specific for a company.

OK. We will close this PR and submit them separately  one by one.
